EKF/ekf_helper.cpp

```diff
@@ -297,67 +303,45 @@ void Ekf::resetHeight()
 	// reset the vertical velocity state
 	if (_control_status.flags.gps && !_gps_hgt_intermittent) {
 		// If we are using GPS, then use it to reset the vertical velocity
-		_state.vel(2) = gps_newest.vel(2);
+		resetVerticalVelocityTo(gps_newest.vel(2));
 
 		// the state variance is the same as the observation
 		P.uncorrelateCovarianceSetVariance<1>(6, sq(1.5f * gps_newest.sacc));
 
 	} else {
 		// we don't know what the vertical velocity is, so set it to zero
-		_state.vel(2) = 0.0f;
+		resetVerticalVelocityTo(0.0f);
 
 		// Set the variance to a value large enough to allow the state to converge quickly
 		// that does not destabilise the filter
 		P.uncorrelateCovarianceSetVariance<1>(6, 10.0f);
-
 	}
 
-	vert_vel_reset = true;
-
 	// store the reset amount and time to be published
 	if (vert_pos_reset) {
 		_state_reset_status.posD_change = _state.pos(2) - old_vert_pos;
 		_state_reset_status.posD_counter++;
 	}
 
-	if (vert_vel_reset) {
-		_state_reset_status.velD_change = _state.vel(2) - old_vert_vel;
-		_state_reset_status.velD_counter++;
-	}
-
 	// apply the change in height / height rate to our newest height / height rate estimate
 	// which have already been taken out from the output buffer
 	if (vert_pos_reset) {
 		_output_new.pos(2) += _state_reset_status.posD_change;
 	}
 
-	if (vert_vel_reset) {
-		_output_new.vel(2) += _state_reset_status.velD_change;
-	}
-
 	// add the reset amount to the output observer buffered data
 	for (uint8_t i = 0; i < _output_buffer.get_length(); i++) {
 		if (vert_pos_reset) {
 			_output_buffer[i].pos(2) += _state_reset_status.posD_change;
 			_output_vert_buffer[i].vel_d_integ += _state_reset_status.posD_change;
 		}
-
-		if (vert_vel_reset) {
-			_output_buffer[i].vel(2) += _state_reset_status.velD_change;
-			_output_vert_buffer[i].vel_d += _state_reset_status.velD_change;
```

Comment:
Is that not required anymore?

Reply:
Thanks. Missed that.

Reply:
@bresch @priseborough  I guess this would also be needed if we reset the vertical velocity in the `resetVelocity()` function. Or is there a reason to apply this only inside `resetHeight()`?

And is there a reason why we add here a delta to the `vel_d` member and below for the delayed and new sample we set it to the new value instead of adding a delta as well?

Reply:
As a general rule the output observer should always be reset following a state reset.